feat(cli): Add ui5 cache clean command (#1455)
SAP/ui5-tooling: internal/documentation/docs/pages/Troubleshooting.md

```sh
ui5 cache clean --yes
```

The command removes the following cached data:
- **UI5 Framework packages** — downloaded UI5 library files (`~/.ui5/framework/`)
- **Build cache (Db)** — build data (`~/.ui5/buildCache/`)

If a previous `ui5 cache clean` was interrupted (e.g. process killed or system crash), the command also detects and removes any leftover data from that interrupted operation, listed as separate entries:
- **Orphaned UI5 Framework packages** — incomplete framework directories left over from a previously interrupted cleanup (`~/.ui5/_framework_to_delete_*/`)
- **Orphaned build cache (Db)** — freed database pages not yet reclaimed during a previously interrupted cleanup

Any required framework dependencies will be re-downloaded during the next UI5 CLI invocation.

::: info
If you have configured a custom data directory via `UI5_DATA_DIR` or `ui5 config set ui5DataDir`, the `ui5 cache clean` command will clean up that location instead of the default `~/.ui5/`. See [Changing UI5 CLI's Data Directory](#changing-ui5-cli-s-data-directory).
:::

::: warning

> Only remove these directories, or run `ui5 cache clean`, when no UI5 CLI process and no `@ui5/*` API consumer is actively running. Running `ui5 cache clean` while `ui5 build` or `ui5 serve` is in progress can break the running process and lead to failed or inconsistent results.

Reviewer: Shouldn't this warning also be printed by the command when the user is asked for confirmation?

Author: And what about if the user proivdes the `--yes` flag (`ui5 cache clean --yes`) ?
Then the command will be executed regardless if there are already running concurent builds / serve.

Maybe, we can include this information also in `ui5 cache --help`. What do you think?

Reviewer: Yes, that would also be helpful. For `--yes` I guess there's no point in showing it.

Author: Done in https://github.com/UI5/cli/pull/1455/changes/0e631fa761826e9d7415eee6594de25f0b1fc07d

| `~/.ui5/buildCache/` | Build cache used by `ui5 build` and `ui5 serve` (see [Build Cache Control](./Builder.md#build-cache-control)) | Yes — rebuilt on next `ui5 build` / `ui5 serve` |
| `~/.ui5/server/` | Locally generated SSL certificate and private key for HTTPS / HTTP/2 mode | Yes — regenerated on next HTTPS server start; the new certificate must be re-trusted |

#### Resolution

> Use the dedicated cache clean command, which removes all cached data:

Reviewer 2: @KlattG could you please do a review here?

```sh
ui5 cache clean
```

For a detailed preview and grouped cleanup summary, use the `--verbose` flag:

```sh
ui5 cache clean --verbose
```

To skip the confirmation prompt (for example in CI environments), use the `--force` flag.
In non-verbose mode with `--force`, the command runs completely silent:

```sh
ui5 cache clean --force
```

The command removes the following cached data:
- **UI5 Framework packages** — downloaded UI5 library files (`~/.ui5/framework/`)
- **Build cache** — build data (`~/.ui5/buildCache/`)

> If a previous `ui5 cache clean` was interrupted (e.g. process killed or system crash), the command also detects and removes any leftover data from that interrupted operation, listed as separate entries:

Reviewer 3: ```suggestion
If a previous `ui5 cache clean` was interrupted (for example, because the process was killed or the system crashed), the command also detects and removes any leftover data from that interrupted operation. This data is listed as separate entries:
```
